plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -1992,20 +1992,50 @@ def move_gear(gear, destination_container, destination_district_uuid, change_dis
         raise OpenShift::UserException.new("Error moving gear. Move gear only allowed from non-districted/districted node to districted node.")
       end
 
+      # get the quota usage and limits from the existing gear
+      gear_quota = get_quota(gear)
+
       destination_node_profile = destination_container.get_node_profile
-      if source_container.get_node_profile != destination_node_profile and app.scalable
-        log_debug "Cannot change node_profile for *scalable* application gear - this operation is not supported. The destination container's node profile is #{destination_node_profile}, while the gear's node_profile is #{gear.group_instance.gear_size}"
-        raise OpenShift::UserException.new("Error moving gear. Cannot change node profile for *scalable* app.", 1)
+      if source_container.get_node_profile != destination_node_profile
+        if app.scalable
+          log_debug "Cannot change node_profile for *scalable* application gear - this operation is not supported. The destination container's node profile is #{destination_node_profile}, while the gear's node_profile is #{gear.group_instance.gear_size}"
+          raise OpenShift::UserException.new("Error moving gear. Cannot change node profile for *scalable* app.", 1)
+        else
+          # get the base quota blocks and files destination container
+          destination_quota = destination_container.rpc_get_facts_direct(["quota_blocks", "quota_files"])
+
+          # set destination container quotas
+          quota_blocks = Integer(destination_quota[:quota_blocks])
+          quota_files = Integer(destination_quota[:quota_files])
+
+          # determine additional storage, if any
+          # subtract source container quota from source node quota
+          # add the difference to destination quota
+          additional_blocks = Integer(gear_quota[3]) - get_quota_blocks
+          additional_files = Integer(gear_quota[6]) - get_quota_files
```

Comment:
I'd be more comfortable if we used source_container.get_quota_blocks and source_container.get_quota_files in the lines above. While the source container is implied to be the current container, it is best not to make this assumption, especially since the source_container is available and explicitly specified in line 1999 above.